targets/TARGET_NORDIC/TARGET_NRF5x/serial_api.c

```diff
@@ -1075,7 +1094,27 @@ void serial_init(serial_t *obj, PinName tx, PinName rx)
  */
 void serial_free(serial_t *obj)
 {
+    MBED_ASSERT(obj);
+
+#if DEVICE_SERIAL_ASYNCH
```

Comment:
The need for this feels very strange.

Is this to say that `obj` can take multiple serial_t types?

Reply:
Yes, all obj across the HAL is like that: https://github.com/ARMmbed/mbed-os/blob/master/hal/serial_api.h#L85-L101